include/nbl/builtin/hlsl/prefix_sum_blur.hlsl

```diff
@@ -0,0 +1,262 @@
+            scanScratchWrapper.base = scanScratch;
+            const float32_t sum = workgroup::inclusive_scan<plus<float32_t>, WorkgroupSize, device_capabilities>::template __call(input, scanScratchWrapper);
+            scanScratch = scanScratchWrapper.base;
+            // loop increment
+            baseIx += WorkgroupSize;
+            // if doing the last prefix sum, we need to barrier to stop aliasing of temporary scratch for `inclusive_scan` and our scanline
+            // TODO: might be worth adding a non-aliased mode as NSight says nr 1 hotspot is barrier waiting in this code
+            if (end + ScanSharedAccessor::Size > SharedAccessor::Size)
+                scratch.workgroupExecutionAndMemoryBarrier();
+            // save prefix sum results
+            if (ix < end)
+                scratch.template set<float32_t>(ix, sum);
+            // previous prefix sum must have finished before we ask for results
+            scratch.workgroupExecutionAndMemoryBarrier();
+        }
+
+        const float32_t last = end - 1;
+        const float32_t normalizationFactor = 1.f / (2.f * radius + 1.f);
+
+        for (float32_t ix = localInvocationIndex; ix < end; ix += WorkgroupSize)
+        {
+            const float32_t result = boxSampler(scratch, ix, radius, borderColor[channel]);
+            data.template set<float32_t>(channel, uint16_t(ix), result * normalizationFactor);
+        }
+    }
+
+    vector<float32_t, DataAccessor::Channels> borderColor;
+    float32_t radius;
+};
+
+template<typename PrefixSumAccessor, typename T>
+struct BoxSampler
+{
+    uint16_t wrapMode;
+    uint16_t linearSize;
+
+    T operator()(NBL_REF_ARG(PrefixSumAccessor) prefixSumAccessor, float32_t ix, float32_t radius, float32_t borderColor)
+    {
+        const float32_t alpha = radius - floor(radius);
+        const float32_t lastIdx = linearSize - 1;
+        const float32_t rightIdx = float32_t(ix) + radius;
+        const float32_t leftIdx = float32_t(ix) - radius;
+        const int32_t rightFlIdx = (int32_t)floor(rightIdx);
+        const int32_t rightClIdx = (int32_t)ceil(rightIdx);
+        const int32_t leftFlIdx = (int32_t)floor(leftIdx);
+        const int32_t leftClIdx = (int32_t)ceil(leftIdx);
+
+        T result = 0;
+        if (rightFlIdx < linearSize)
+        {
+            result += lerp(prefixSumAccessor.template get<T, uint32_t>(rightFlIdx), prefixSumAccessor.template get<T, uint32_t>(rightClIdx), alpha);
+        }
+        else
+        {
+            switch (wrapMode) {
+            case ETC_REPEAT:
+            {
+                const T last = prefixSumAccessor.template get<T, uint32_t>(lastIdx);
+                const T floored = prefixSumAccessor.template get<T, uint32_t>(rightFlIdx % linearSize) + ceil(float32_t(rightFlIdx % lastIdx) / linearSize) * last;
+                const T ceiled = prefixSumAccessor.template get<T, uint32_t>(rightClIdx % linearSize) + ceil(float32_t(rightClIdx % lastIdx) / linearSize) * last;

```

Comment:
something modulo `N-1` and then divided by `N` and then ceiled will ALWAYS return 1, unless the modulo was 0

this calculation must be wrong, also the overlap counting part is common to both inputs to `lerp` so can be hoisted out of it.

its only when floored is N-1 and ceiled is 0 after modulo, you have a wraparound, and need to add a `last` to the `ceiled` to compensate

and independently of that you add `(rightFlrIdx/N)*last` to the overall interpolated thing